src/main/java/duke/command/DoneCommand.java

```diff
@@ -0,0 +1,30 @@
+package duke.command;
+
+import duke.main.*;
+import duke.exception.DukeException;
+import duke.task.*;
+
+public class DoneCommand implements Command {
+    private int index;
+
+    public DoneCommand(int index) {
+        this.index = index;
+    }
+
+    public void execute(Storage storage, Ui ui, TaskList tasks) throws DukeException {
+        try {
+            Task doneTask = tasks.getTask(index);
+            doneTask.markDone();
+            ui.output("Nice! I've marked this task as done: \n" + doneTask.toString());
+            storage.writeToFile(tasks);
+        }
+        catch (Exception e) {
```

Comment:
We should specify which exception we are catching (DukeException or IOException in this case) and not use the base class Exception to catch all exceptions